File: SHOOTRZ/backend/utils/smoothing.py
from typing import List


def moving_average(series: List[float], window: int = 5) -> List[float]:
    if window <= 1 or len(series) == 0:
        return series
    out = []
    for i in range(len(series)):
        start = max(0, i - window + 1)
        window_vals = series[start : i + 1]
        out.append(sum(window_vals) / len(window_vals))
    return out

File: SHOOTRZ/src/utils/smoothing.ts
export function movingAverage(series: number[], window = 5): number[] {
  if (window <= 1 || series.length === 0) return series
  const out: number[] = []
  for (let i = 0; i < series.length; i++) {
    const start = Math.max(0, i - window + 1)
    const slice = series.slice(start, i + 1)
    out.push(slice.reduce((a, b) => a + b, 0) / slice.length)
  }
  return out
}








